jwst/extract_1d/extract1d.py

```diff
@@ -452,6 +454,9 @@ def _fit_background_model(image, x, j, bkglim, bkg_fit, bkg_order):
     wht = wht[good]
     y = y[good]
 
+    if wht.sum() == 0:
```

Comment:
Reviewers: This is the only substantive change to fix the bug. All other changes are purely cosmetic.

Reply:
Do you think a log message/warning is in order, or should this effectively never run unless the data aren't reasonable?

Reply:
Yes, both here and in line 450 (where npts is insufficient) this function returns 0 for the number of bkg region points used, which in turn triggers a warning message in the calling function.

Reply:
Ah, I didn't do due diligence on tracing what the zeroed-out polynomial fit would cause downstream. Sounds good!